### DISCUSSION TITLE
Question about recording touchscreen

### DISCUSSION THREAD

**Discussion by maxmckenz** [UmerCodez/SensorServer#68] [Q&A]
Hi!

Thank you so, so much for this excellent piece of software - it works great! I have a question about recording touch-screen inputs.

I'm a researcher who is working on the influence of turbulence on touchscreen interaction in aircraft. My goal is to have a setup where I can project a symbol (like a dot) on the screen in a circular movement. At the same time, I would like to record the touch location as the user tries to follow the dot around the circle. This would then allow us to create models of people's arms as they get perturbed by turbulence in an aircraft.

My question is the following: is it only possible to record touchscreen data when you open the "Touch Screen" tab in the application, or are there other options? What would the easiest solution look like for my goal?

Thanks so much!

Cheers,
Max

**Comment by UmerCodez** [comment]
Hi Max,  

Thank you for your kind feedback!  

If you’re developing an Android app for this purpose, where users follow the circular movement of a dot with their touch, you can easily record touchscreen data using the Android's `MotionEvent` API. This API allows you to track touch interactions such as `ACTION_DOWN`, `ACTION_MOVE`, and `ACTION_UP`, providing precise coordinates of the user’s touch. However, if you are using the SensorServer app, it only sends touchscreen events over the network when the user interacts with the touchscreen via the "Touch Screen" tab. Alternatively, you can access raw touchscreen data using the [`adb`](https://developer.android.com/tools/adb)'s command `adb shell getevent -l`. This command logs all touch events directly from the Android device, regardless of which app is open or in use.
